### PR TITLE
Add scalebar to ASA

### PR DESCRIPTION
**Pull request by dgboss**
- adds a scalebar to the ASA map
- fixes navigation controls through inclusion of default OpeenLayers css
- I'm not entirely happy with the implementation. At certain zoom levels the distance indicators on the scale bar overlap. This is due to the addition of padding to get the white background. Internally the control calculates its size and updates via css, so the addition of padding messes things up a little.
# Test Links:
[Landing Page](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3501.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 1]
```diff
@@ -353,6 +364,36 @@ const FBAMap = (props: FBAMapProps) => {
...
+            ['& div.ol-scale-bar-inner div>div.ol-scale-step-marker']: {
+              top: '-8px !important'
+            },
+            ['& div.ol-scale-step-text']: {
+              bottom: '5px !important'
+            }
+          }}
+          ref={scaleRef}
+        ></Box>
+      </Box>
```
Comment (dgboss) [inline web/src/features/fba/components/map/FBAMap.tsx:396]: I tried creating a component out of this but got blocked with ref forwarding and integration with the OpenLayers' ScaleLine control which needs to have its target set to a ref.

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 2]
```diff
@@ -284,6 +287,14 @@ const FBAMap = (props: FBAMapProps) => {
     })
     mapObject.setTarget(mapRef.current)
 
+    const scaleBar = new ScaleLine({
+      bar: true,
+      minWidth: 160,
+      steps: 5
```
Comment (brettedw) [inline web/src/features/fba/components/map/FBAMap.tsx:293]: Looks good! Would changing this to 4 solve the overlap issue?

Comment (dgboss) [reply web/src/features/fba/components/map/FBAMap.tsx]: Great suggestion! It fixed the problem by reducing the amount of text. Thanks!

**Review by brettedw — APPROVED** [review]

File: web/src/features/fba/components/map/ScaleBarContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      }}
+    >
+      <Box
+        ref={ref}
+        sx={{
+          ['& div.ol-scale-bar']: {
+            position: 'static'
+          },
+          ['& div.ol-scale-bar-inner div.ol-scale-step-marker']: {
+            top: '13px !important'
```
Comment (dgboss) [inline web/src/features/fba/components/map/ScaleBarContainer.tsx:25]: The `!important`s are ugly but necessary to override the OpenLayers styles which are declared on the actual HTML elements, not just in css classes.

**Review by conbrad — APPROVED** [review]
Do you think we can adjust how to display the map to avoid some of the pitfalls mentioned? Map controls also seemed to be affected by the way we use the grid layout on the page.

**Pull request merged**